### ISSUE TITLE
Inserting plots from csv files

### ISSUE THREAD

**Issue by cserteGT3** [tlienart/JuDoc.jl#182]
Continuing the discussion started on [discourse](https://discourse.julialang.org/t/ann-judoc-jl-a-static-site-generator-in-julia/23693/14).
> On master you can now also directly evaluate Julia code blocks on the fly similar to weave which is what it looks like you want to do, the only hiccup is that the generation of the page where the plot is will be slowed down significantly by the drawing of the plot due to time to first plot issue.

I am aware of the time-to-first-plot issue and in this case I don't have any problems with it.

I've created an [example repo](https://github.com/cserteGT3/JuDocPlottest), where I included your example too. I added an example `.tsv` file, and pages generated by different packages: JuDoc itself, Literate, Weave (to html) and Weave (to markdown). For both weave output the same (`.jl`) source file is used (it's possible that the outputs could be optimized with different source files).

To sum up my results:
JuDoc:
* If I hide all the lines of a source block, a blank inlined code block will be placed.
* I can't inject the output to the page like raw text (and not inlined code block).

Literate:
* I managed to do everything I wanted.

Weave (to html):
* Works as expected: the generated page is fine,, but the definitons (`@def`) stay.

Weave (to markdown):
* I managed to do everything I want.

The output of Literate and Weave (to md) look the same (as expected).

If I had to choose from the current solutions, I would probably go with Literate. I would choose JuDoc if we can solve the above mentioned problems.

**Comment by tlienart** [comment]
Hello! thanks for the description and comparison. On master there are now two new features that should help:

1. you can use `#hideall` somewhere in. a code block that you don't want to display, this has the same impact of hiding each line. Also thanks for reporting the empty code block problem, that should be fixed now.
1. if you use. `\textoutput{...}` instead of `\output{...}`, the output will be processed as markdown. 

This example should show this in action:

```markdown
@def hascode = true

Code block:

```julia:./test
#hideall
temps = (15, 23, 20, 15, 20, 17, 18, 30, 21, 19, 17)
a = sum(temps)/length(temps)
println("The _average_ temperature is **$(round(a, digits=1))°C**.")
```

\textoutput{./test}

The end.
```

(the figure is as before, that didn't seem to be a problem)

Feedback is welcome! (also for the command naming btw, though you can always re-define your own commands for better naming).

**Comment by cserteGT3** [comment]
Thank you!
I've updated the repo with your answers.

**Comment by tlienart** [comment]
awesome, I'll close this for now but please re-open an issue if you encounter other problems or have ideas on how to improve this stuff
